### ISSUE TITLE
Code Updates

### ISSUE THREAD

**Issue by alnopa9** [haggis78/BreconChurch#14]
Hello Brecon Team, 
So, it's been bothering me that the drop downs in your nav bar wouldn't work. Good news though! I figured out the problem. Long story short, the javascript that enables the nav bar to be 'sticky' changes the class for that div from 'navbar' to 'sticky'. This means that the css that was assigned to the 'navbar' doesn't fire now. To fix this, I nested the navbar into another div that fires the javascript.
I'm pushing up the fixed code to your repo now.

**Comment by alnopa9** [comment]
Just pushed the bug fix! [commit](https://github.com/haggis78/BreconChurch/commit/b49e13b1d8fd1eca7a0b3f0b74094a3e40d486f0)

**Comment by alnopa9** [comment]
Also, made the header image (brecon.png) into an svg (brecon.svg) and updated the files to use the .svg file extension. [commit](https://github.com/haggis78/BreconChurch/commit/967f167dfc4cb21ae7de152d1fb6daafd3c5cd48)

**Comment by alnopa9** [comment]
# Jumping with Nav Bar
I noticed with the site that the navbar 'jumps' to the top of the page on 'texts.html'. This was an easy fix. In the css, there is a class called 'content' which holds everything under the navbar. Nest everything, except the inline javascript, in this class. [commit](https://github.com/haggis78/BreconChurch/commit/490bba02c06e44b0b1d328d1a672ab0d1af10411)

@amberpeddicord the server will need updated.

**Comment by alnopa9** [comment]
Brecon team, the main points in my latest pushes are that I made the sticky function javascript it's own file that works! That means no more inline javascript. (YAY!! 🎉 ) I also updated the navbar to include two new drop downs; one for reading each manuscript and one that holds the comparison versions of each manuscript.

[sticky push](https://github.com/haggis78/BreconChurch/commit/3fb84ef900c53eb4204d65540d7f452747c1b7a4)

[drop downs push](https://github.com/haggis78/BreconChurch/commit/a40f213bc5e9e979d354772589e8372ae7fdb184)

I am currently trying to sort out the manuscript XSLT so that it reads the distinct-values() for the manuscript editions and outputs a full print for each one.

**Comment by ebeshero** [comment]
@alnopa9 yay!!! What did you need to do to get the separate JS to work? Thanks for taking on this advanced XSLT and JS challenge for the group, and let me know if you want help w `<xsl:result-document>`.

**Comment by amberpeddicord** [comment]
@alnopa9 Thank you so much for your help on our project! We really appreciate it!

**Comment by alnopa9** [comment]
@ebeshero I just had to wrap the window.onscroll function in a window.onload function.

**Comment by alnopa9** [comment]
Brecon Team, the latest push includes a new XLST, [full-collation-2](https://github.com/haggis78/BreconChurch/blob/master/full-collation-2.xsl). It outputs a display and comparison version of each manuscript edition. However, it's very brittle and long because it uses `<xsl:result-document>` to do so and relies on a mode for each type and edition. @ebeshero I have another version trying to use distinct-values() of the manuscript editions, but I kept getting the same error. [commit](https://github.com/haggis78/BreconChurch/commit/9c1de604ab242cfa1c55b2ba62dedbf2a5fd6f47)

**Comment by ebeshero** [comment]
@alnopa9 Let me see if I can debug the simpler one in your commit...

**Comment by ebeshero** [comment]
@alnopa9 ...and I'll work on it after class / this evening! What you're producing is similar to what I've been coding for the Frankenstein Variorum project!

**Comment by alnopa9** [comment]
@ebeshero What I've been working on. 

[commit](https://github.com/haggis78/BreconChurch/commit/ab62d8b0f64b99634c3fda67a26725413a9b7566)
[Collation XSLT](https://github.com/haggis78/BreconChurch/blob/master/xslt/full-collation.xsl)

**Comment by ebeshero** [comment]
@alnopa9 Okay! I'm pulling it in now and I'll see if I can get it working (and add comments as I go).

**Comment by alnopa9** [comment]
@ebeshero I've made some headway, I just have to get the one template rule to work now.

**Comment by ebeshero** [comment]
@alnopa9 Want me to wait for another commit?

**Comment by alnopa9** [comment]
@ebeshero I pushed to fix the javascript links. You can push yours up.

**Comment by ebeshero** [comment]
Okay...I'll work on what I have here currently, @alnopa9 !

**Comment by ebeshero** [comment]
@alnopa9 Aha! I know what the issue is...We need to pass a parameter to hold the value of current through your template matches...Yay! Stay tuned and I'll push up the code...

**Comment by ebeshero** [comment]
Okay! I just pushed up some revised XSLT and output with [the latest commit](https://github.com/haggis78/BreconChurch/commit/328052ca3b9387c811eb6708fe8cf668d8162412)! Pull it in and have a look. I can see that at least some of the editions are getting your `<span class="variance">` elements now...

**Comment by ebeshero** [comment]
@alnopa9 I'm going to add some comments to this and push again. I want to explain how `<xsl:with-param/>` and `<xsl:param>` work. They are basically *local* variables that get passed along from template to template. We can't declare your `current()` node as a global variable, since your template that matches on the document node is recursing through a set of the editions. So, we pass the `current()` edition node down to the other template rules using `<xsl:with-param>` inside `<xsl:apply-templates>` and at the start of each new template using `<xsl:param>`.  It reminds me of tunneling somehow.

**Comment by alnopa9** [comment]
@ebeshero Okay! I'm looking it over and it does make sense why that works.

**Comment by ebeshero** [comment]
@alnopa9 I just pushed a version with my comments on how xslt parameters work! See [the latest commit](https://github.com/haggis78/BreconChurch/commit/506e13acc2bdfdca2bf905c595d9f17c8b12bfd5). :-)

**Comment by alnopa9** [comment]
@ebeshero Thanks! I am going to use this to help add on the comparison editions of the manuscript now!
Edit:
I included in the same xslt another `<xsl:result-document>` to output the comparison editions as well. [commit](https://github.com/haggis78/BreconChurch/commit/9e6e6e62c46d71166c3d7ccb2a253e315b782dd7)

**Comment by alnopa9** [comment]
Leaving these links here for reference later when we allow synchronized scrolling on the three comparison editions
[Synchronized Scrolling Stack Overflow Issue (JQuery)](https://stackoverflow.com/questions/31524371/how-to-synchronize-scroll-between-two-windows)
[Synchronized Scrolling Fiddle (JQuery)](http://jsfiddle.net/Lv2dw787/4/)

[Pure JavaScript Synchronized Scroll S.O. Issue](https://stackoverflow.com/questions/9236314/how-do-i-synchronize-the-scroll-position-of-two-divs)
[Pure JavaScript Synchronized Scroll Fiddle](https://jsfiddle.net/2brahp0m/)

Update: Overthought this strategy. It's actually a lot easier. Going to push up a rough draft for the comparison page. Also linking this [W3Schools Tutorial for making text appear with a checkbox](https://www.w3schools.com/howto/howto_js_display_checkbox_text.asp)

Update:
Just pushed the rough comparison page with editions C, I, and R (I think). I need to fix the links in the nav bar, they are broken at the moment. Also updated the editions html pages to include the latest collation for O.
[commit](https://github.com/haggis78/BreconChurch/commit/a2c8284ba51238f1cc211fb5ec3594ec4b1dff01)
[Comparison Page](https://github.com/haggis78/BreconChurch/blob/a2c8284ba51238f1cc211fb5ec3594ec4b1dff01/site/html/comparison.html)

**Comment by alnopa9** [comment]
@haggis78 and Brecon team, I have good news!! I got the javascript for the buttons to work on the comparison page! I'm about to push everything up now. If you open the comparison.html file locally, you should be able to play around with it! It doesn't line up perfectly yet, but the next step would be to write an xslt that would produce the javascript file and the comparison file so you don't have to manually edit the pages as much. Tell me if you like the button layout.
[commit](https://github.com/haggis78/BreconChurch/commit/7386a15dd3ea76ced7d31bc96f35ce0f01239820)
[Comparison Page](https://github.com/haggis78/BreconChurch/blob/master/site/html/comparison.html)

Update:
I just added a simple color change to the text that has variance class on it to help highlight the differences between editions. [commit](https://github.com/haggis78/BreconChurch/commit/23222661ce55775e812e95e84fd9239cd8b427ee)

**Comment by alnopa9** [comment]
@amberpeddicord I noticed that the css is not linked properly on the server. I think this is because the repo was reorganized. I would clear the server and upload all of the contents in the site folder. (but not the site folder itself.) I just moved the index.html file out of html and into site to accommodate for this change.

**Comment by alnopa9** [comment]
@amberpeddicord Can you update the server? I just updated a lot of the pages. [commit](https://github.com/haggis78/BreconChurch/commit/58df066a5da9ad6efc330ca28921a57677f1eddf)

**Comment by amberpeddicord** [comment]
@alnopa9 Updated! Thank you so much!

**Comment by alnopa9** [comment]
@amberpeddicord I fixed some minor bugs. Can you update the server again?

**Comment by amberpeddicord** [comment]
@alnopa9 Updated!

**Comment by alnopa9** [comment]
@amberpeddicord Okay, I think this will be the last one tonight. Can you update the server again?

**Comment by alnopa9** [comment]
@amberpeddicord I added a toggle switch to the display manuscripts on the site. There is a new javascript file on those pages, the css has been updated, and the ssi needed changed as well. When you get a chance can you update the server? [commit](https://github.com/haggis78/BreconChurch/commit/7c61e4f17e8b236be2aba73c2a82b72abb88ed0f)

**Comment by alnopa9** [comment]
@haggis78 I got the word count to work in my column svg. The word count is located directly under the AB header for each block, and the numbers underneath that count string lengths for each app in the AB. You'll have to open it locally to view it because it is an html file.
[commit](https://github.com/haggis78/BreconChurch/commit/ddd2e3eb2158f1b15876cfe47a86666e1b23b45c)

**Comment by alnopa9** [comment]
Brecon Team, I updated the svg for the edition-svg to now be more compact. Instead of each edition having a whole column to itself, they now have a column that is a certain color. Another addition is the hover effect on the svg. Now, when you hover over any of the string lengths in the AB blocks, the text will display as a tool tip.
[commit](https://github.com/haggis78/BreconChurch/commit/8c0dc1dbd4b51e32bf60a1c180cbebb54c7c09d6)
[edition-svg.html](https://github.com/haggis78/BreconChurch/blob/master/site/html/edition-svg.html)

**Comment by alnopa9** [comment]
Brecon Team, I got the ab-bar-graph xslt to work! It produces a bar graph for each AB and displays the word counts for each manuscript.
[commit](https://github.com/haggis78/BreconChurch/commit/83214be0071043f435a705732da2508655d37e2d)
[edition-bar-graph-svg.html](https://github.com/haggis78/BreconChurch/blob/master/site/html/edition-svg-bar-graph.html)

**Comment by ebeshero** [comment]
For those who can’t wait to do a git pull to see it in the browser, here it is with the GitHub htmlpreview prefix:

http://htmlpreview.github.io/?https://github.com/haggis78/BreconChurch/blob/master/site/html/edition-svg-bar-graph.html

**Comment by alnopa9** [comment]
Brecon Team, I have updated the word count bar graph to include a title and finished the bar graph for the string counts.
[commit](https://github.com/haggis78/BreconChurch/commit/602ea1d573e0acdaf10de6c7a792137b936feb6b)
[Word Count Bar Graph](https://github.com/haggis78/BreconChurch/blob/master/site/html/edition-svg-word-count-bar-graph.html)
[String Count Bar Graph](https://github.com/haggis78/BreconChurch/blob/master/site/html/edition-svg-string-count-bar-graph.html)

The next thing on my to-do list is update the AB Comparison page to include those two bar graphs.

**Comment by haggis78** [comment]
@alnopa9 Awesome. FYI in my editing over the weekend I subdivided some of the `<ab>`s, because that turned out to be the consensus of the manuscripts. I assume that your code is written to automatically adapt to the `<ab>`s it finds, wherever they are?

**Comment by alnopa9** [comment]
@haggis78 Yes, I can run everything through the XSLTs again to make them up to date.

**Comment by alnopa9** [comment]
Brecon Team, I finished the [ab comparison page](https://raw.githubusercontent.com/haggis78/BreconChurch/master/site/html/edition-svg.html). It's a bit lengthy, so I had to link to the raw view of the code. There are now two check boxes in the header (to the right of the comparison drop down), one to make the string count bar graphs visible and one to make the word count bar graphs visible. The only problem with this is that the bar graphs are not in line with their appropriate ABs. I'm looking in to how to fix that now.
[commit](https://github.com/haggis78/BreconChurch/commit/ef7c2d89f9567de7dc2386f5c6e83362bb84f319)
